src/api/transaction.ts

```diff
@@ -355,15 +346,44 @@ export class Transaction {
    *
    * @return PendingTransactionResponse
    */
-  async signAndSubmitTransaction(args: {
-    signer: Account;
+  async signAndSubmitTransaction(
+    args: FeePayerOrFeePayerAuthenticatorOrNeither & {
+      signer: Account;
+      transaction: AnyRawTransaction;
+    },
+  ): Promise<PendingTransactionResponse> {
+    return signAndSubmitTransaction({
+      aptosConfig: this.config,
+      ...args,
+    });
+  }
```

Comment:
This is very niche, is it just so when you're running your own fee payer, and you custody keys for users?

Thoughts on this @0xmaayan ?

Reply:
My goal really was just to eliminate the extra step that people sign and then submit.  In the niche case where someone runs their own fee payer or custody for users, they can also use this by providing both accounts.

But mainly I just wanted an API where you have the authenticator from the other party (either as the sender or as the feepayer), and then you can just sign and submit

Reply:
I am abit confused, why we need these changes if we introduce signAndSubmitAsFeePayer?

IMO, the new function arguments is confusing

Reply:
So signAndSubmitAsFeePayer requires access to signing capabilities as the feePayer and the authenticator already signed by the sender.  Then the feepayer is left to sign and submission can be done by the feepayer (if they want to).

Alternatively if the sender asks a feepayer to sign and return an authenticator, this allows for them to use signAndSubmit. 

In the case that the sender and feepayer are colocated  then they can use signAndSubmit.  

Right now the paradigm for fee payer
1. get sender authenticator
2. get feepayer authenticator
3. submit via `await aptos.transaction.submit.simple`

I want to eliminate use of await aptos.transaction.submit.simple

I think in vast majority of cases people sign and submit immediately after.  So they should use signAndSubmit.

This is a better devX for Keyless because we support asyncronous proof fetching and it's just simpler.

Reply:
I think this generally is going to give more confusion, but I'm not going to fight this one too long.